firmware/ubirch/m66/m66_tcp.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * M66 TCP/IP operations.
+ * Simple driver with send and receive functions
+ *
+ * @author Niranjan H. Rao
+ * @date 2016-10-07
+ *
+ * @copyright &copy; 2015 ubirch GmbH (https://ubirch.com)
+ *
+ * ```
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ```
+ */
+#include "m66_tcp.h"
+#include "m66_core.h"
+#include "m66_ops.h"
+#include "m66_parser.h"
+#include "ubirch/timer.h"
+#include <stdio.h>
+
+bool modem_mqtt_connect(const char *apn, const char *user, const char *password, uint32_t timeout) {
+  timer_set_timeout(timeout * 1000);
+
+  if (!modem_gprs_attach(apn, user, password, uTimer_Remaining)) return false;
+
+//  modem_send("AT+QIMUX=0");
```

Comment:
please add the comment what the commented out part does, or remove it if we will never use it